Add Deeper_Network Open Grant Milestone Delivery document

fyang1024 opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/581

mmagician commented:
@fyang1024 Thanks for the delivery, we'll check it as soon as we can!

mmagician commented:
@fyang1024 We noticed that you mention:
> Approved for grant from Web3 Foundation

On your [one-pager](https://deeper.network/onepager).

Please note that you are not allowed to use endorsement from the Grants Program until *after* at least one of your milestones has been approved and merged, as per our [guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant-badge-guidelines.md).

I would like to kindly ask you to remove that information from your website please.

fyang1024 commented:
> @fyang1024 We noticed that you mention:
> 
> > Approved for grant from Web3 Foundation
> 
> On your [one-pager](https://deeper.network/onepager).
> 
> Please note that you are not allowed to use endorsement from the Grants Program until _after_ at least one of your milestones has been approved and merged, as per our [guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant-badge-guidelines.md).
> 
> I would like to kindly ask you to remove that information from your website please.

fyang1024 commented:
@mmagician Sorry about that I didn't check every single doc published. They mistaken grant application being accepted with grant being approved. It's corrected now. Please have a look at the [one-pager](https://deeper.network/onepager) again.

mmagician commented:
@fyang1024 Thanks for this, I appreciate it.

mmagician commented:
@fyang1024 We noticed that you're still using the mention of the Web3 Foundation Grant on your homepage. For the same reasons as outlined above, please remove that mention. Also, clicking on that banner shows text which states:
> [...] handpicked by the Parity committee over numerous applications [...]. The milestone partnership marks [...] the pivotal role in the Polkadot ecosystem. [...] With this partnership in place [...]

First of all, you don't know how many other applications there were to SBP. Furthermore, this is not a partnership, neither with the Web3 Foundation nor Parity. Lastly, you shouldn't call yourselves having a "pivotal role" - this is something subjective and you haven't even launched on Polkadot yet. What you are stating on your website is simply not true. 

Please remove all of the above mentions, and I recommend to double check with relevant parties (Parity/W3F) before using any kind of related information in the future.

fyang1024 commented:
@mmagician Sorry again for that being overlooked. It's a shame even though I was not involved the website design. Now it's cleared. I just checked the whitepaper and found nothing related to the Open Grant. That's all the public docs I can see now.

semuelle commented:
@fyang1024, could you provide access to https://github.com/atomos-dev/control-plane?

semuelle commented:
@fyang1024, the micropayment tests fail on their second run. Also, [these](https://github.com/deeper-chain/deeper-chain/blob/9449a367a85bfaba2232945a7aa6d8ec157ab68c/scripts/micropayment_test.js#L148-L175) tests already fail on the first run. Looks like the claims are off by a digit. Can you verify?

```
❯ node micropayment_test.js
start wsprovider...
hehe err: null
hehe api: [object Object]
OpenChannel [Alice -> Bob]
total channel balance = 100, test OK
OpenChannel [Alice -> Charlie]
total channel balance = 200, test OK
OpenChannel [Alice -> Dave]
total channel balance = 300, test OK
ClaimPayment [Alice-> Bob]
nonce: 0, session_id: 1, amt: 10, signature: f6fe34016e38dfb5ca68145807f810ba79432b248be191217809c377958fbd2f986eb4a3745d99f55c31f253fce3feef417f95a1d6fadad2bd5dc4be050f0f8b
total channel balance = 200, test Failed
close channel [Alice -> Charlie]
total channel balance = 100, test Failed
addBalance to channel [Alice -> Dave]
total channel balance = 130, test Failed
total channel balance = 0, test OK
```

hliu4atomos commented:
@semuelle  
Hi  Semuelle, the invitation has been sent, please check your email. Thanks.

fyang1024 commented:
> @fyang1024, the micropayment tests fail on their second run. Also, [these](https://github.com/deeper-chain/deeper-chain/blob/9449a367a85bfaba2232945a7aa6d8ec157ab68c/scripts/micropayment_test.js#L148-L175) tests already fail on the first run. Looks like the claims are off by a digit. Can you verify?
> 
> ```
> ❯ node micropayment_test.js
> start wsprovider...
> hehe err: null
> hehe api: [object Object]
> OpenChannel [Alice -> Bob]
> total channel balance = 100, test OK
> OpenChannel [Alice -> Charlie]
> total channel balance = 200, test OK
> OpenChannel [Alice -> Dave]
> total channel balance = 300, test OK
> ClaimPayment [Alice-> Bob]
> nonce: 0, session_id: 1, amt: 10, signature: f6fe34016e38dfb5ca68145807f810ba79432b248be191217809c377958fbd2f986eb4a3745d99f55c31f253fce3feef417f95a1d6fadad2bd5dc4be050f0f8b
> total channel balance = 200, test Failed
> close channel [Alice -> Charlie]
> total channel balance = 100, test Failed
> addBalance to channel [Alice -> Dave]
> total channel balance = 130, test Failed
> total channel balance = 0, test OK
> ```

hi  @semuelle , thank you for the findings. You're right. Those 3 tests failed, but fortunately it's not due to the chain's bug.  The claim payment test failed because the channel expired and hence a claim triggered channel closure. The channel expired too early because a wrong parameter value 1 was supplied as duration (in seconds) to the `open_channel` function calls. I think the duration param was in eras instead of seconds at one stage, and later on we changed it to seconds for more flexibility. And our CI only runs unit tests but not the integration tests, therefore this failed test was never picked up. 
I have fixed the test and did some refactoring and cleanup as well https://github.com/deeper-chain/deeper-chain/commit/285d61c52cd3751145b114b271213b2f0ba59fe2

Now the tests can pass, and they can run multiple times without problem. Please have another try. 

Thank you

AmadeusGB commented:
@semuelle 
hi semuelle，This is a question I wish to explain to you.

First, the micropayment pallet currently contributed to the Web3 grant is a service deployed independently on the blockchain.

Secondly, the deeper layer1 is designed for external calls and does not have any code coupling with the micropayment pallet.

Third, the code of control plane has many codes and keys related to the security of our device, in addition to external calls to micropayment, so the overall open source will have security risks. Can we communicate on this part to review in a private environment or open source only the control plane code related to micropayment?

semuelle commented:
Hi @AmadeusGB, the micropayment related code is enough. In fact, it would suffice to publish a minimal example of the client-to-blockchain calls and client-to-client transactions. It's just so others can quickly duplicate and verify your results/setup.

AmadeusGB commented:
Thank you very much. 
Web3 grant is a great platform. We will continue to work hard and hope to make greater contributions to the Polkadot ecology.

AmadeusGB commented:
hi @RouvenP we have received it, thank you very much!
